src/components/Inventory/view/BatchesCellRenderer.tsx

```diff
@@ -41,29 +52,102 @@ export default function BatchesCellRenderer(props: RendererProps<TableRowType>):
     return <Typography className={classes.text}>{iValue}</Typography>;
   };
 
+  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+    stopPropagation(event);
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  const openChangeQuantityHandler = (type: string) => {
+    handleClose();
+    setModalValues({
+      openChangeQuantityModal: true,
+      type,
+    });
+  };
+
   if (column.key === 'withdraw') {
+    return (
+      <>
+        <CellRenderer
+          index={index}
+          column={column}
+          row={row}
+          value={
+            <Button
+              id='withdraw-batch'
+              label={strings.WITHDRAW}
+              onClick={rowClick}
+              size='small'
+              priority='secondary'
+              className={classes.text}
+            />
+          }
+        />
+      </>
+    );
+  }
+
+  if (column.key === 'batchNumber') {
+    return <CellRenderer {...props} value={createLinkToBatch(value)} />;
+  }
+
+  if (column.key === 'quantitiesMenu') {
     return (
       <CellRenderer
         index={index}
         column={column}
         row={row}
         value={
-          <Button
-            id='withdraw-batch'
-            label={strings.WITHDRAW}
-            onClick={rowClick}
-            size='small'
-            priority='secondary'
-            className={classes.text}
-          />
+          <>
+            <ChangeQuantityModal
```

Comment:
feels like this should be a separate component?
functions like handleClick and setAnchorEl in batches cell renderer could be renamed to something more specific (since it pertains to the change quantity features).